### PR TITLE
Create a new renderer for each new plot type

### PR DESCRIPTION
**Pull request by sankhesh**
Currently, the test whether the renderer exists is not sufficient
to decide whether a new renderer needs to be created. This was
causing different plots to use the same renderer.

This change introduces a new argument to the fitToViewport call
that forces it to create a new renderer if requested. This will
enable all the pipeline plot calls to create their own
independent renderer.

Fixes issue #1338

### CODE REVIEW

**Comment by sankhesh** [comment]
@doutriaux1 @aashish24 @dlonie Please review

**Comment by doutriaux1** [comment]
@sankhesh considering you had to change 6 files with the same edit, shouldn't fit2viewport be moved to piepline2d? (or even pipeline?)

**Comment by doutriaux1** [comment]
@sankhesh will apporve this as soon as @aashish24 approved https://github.com/UV-CDAT/uvcdat/pull/1418 otherwise merge conflicts might arise and that will delay @aashish24 reviewing of it even further.

**Comment by sankhesh** [comment]
There are some other places that fitToViewport is called. I'm not sure if all of that can be moved. Plus, we might be able to get rid of fitToViewport once the Context2D work goes in.

**Comment by sankhesh** [comment]
Baseline for new test added in UV-CDAT/uvcdat-testdata@ca1be96

**Comment by sankhesh** [comment]
@doutriaux1 

This change causes one test failure on my linux machine. It is the `text_vcs_basic_boxfill_masked_aeqd_proj` test.

With my changes I get the following output image:
test_vcs_basic_boxfill_masked_aeqd_proj
 

Whereas the test image looks like:
test_vcs_basic_boxfill_masked_aeqd_proj

Basically, now more area of the plot is visible and un-masked. Since, I haven't touched any masking code, I am inclined to think that the original baseline was wrong and these changes produce the right image. Do you concur? If yes, I will update the baseline.

**Comment by doutriaux1** [comment]
@sankhesh this plot is worng in both version anyway.... Look at Africa it should have purple blob over sahara but it only has 1/2 the blob. This is another wrap issue with projections that we should look into. Let's update the baselines and record an issue for this "bad" plot.

**Comment by sankhesh** [comment]
Thanks for quick reply @doutriaux1. Will do.

**Comment by sankhesh** [comment]
@doutriaux1 

Projection issue documented here: #1462

**Comment by aashish24** [comment]
@sankhesh can you update this branch so that we can merge it soon? Thanks.

**Comment by sankhesh** [comment]
@aashish24 Fixed merge conflicts

File: Packages/vcs/Lib/VTKPlots.py [hunk 1]
```diff
@@ -1033,7 +1037,8 @@ def scaleLogo(self):
                 self.setLayer(self.logoRenderer, 1)
                 self.renWin.AddRenderer(self.logoRenderer)
 
-    def fitToViewport(self, Actor, vp, wc=None, geo=None, priority=None):
+    def fitToViewport(self, Actor, vp, wc=None, geo=None, priority=None,
```
Comment (aashish24) [inline Packages/vcs/Lib/VTKPlots.py]: the style seems off or may be the github is not showing me the right diff?

Comment (sankhesh) [inline Packages/vcs/Lib/VTKPlots.py]: What do you think is wrong here?

**Comment by sankhesh** [comment]
Continued in #1468

**Pull request closed without merging**